lib/benchee/formatters/json.ex

```diff
@@ -55,84 +59,59 @@ defmodule Benchee.Formatters.JSON do
+      ...>            median:        450.0,
+      ...>            minimum:       200,
+      ...>            maximum:       900,
+      ...>            mode:          nil,
+      ...>            sample_size:   8,
+      ...>            percentiles:   %{99 => 900}
       ...>          }
       ...>        },
       ...>      ]
       ...>    }
       iex> Benchee.Formatters.JSON.format(suite, %{file: "my_file.json"})
-      %{"Some Input" => "{\\"run_times\\":{\\"My Job\\":[200,400,400,400,500,500,700,900]},\\"sort_order\\":[\\"My Job\\"],\\"statistics\\":{\\"My Job\\":{\\"average\\":500.0,\\"ips\\":2.0e3,\\"maximum\\":900,\\"median\\":450.0,\\"minimum\\":200,\\"mode\\":400,\\"percentiles\\":{\\"99\\":900},\\"sample_size\\":8,\\"std_dev\\":200.0,\\"std_dev_ips\\":800.0,\\"std_dev_ratio\\":0.4}}}"}
+      "[{\\"input_name\\":\\"Some Input\\",\\"memory_usage_statistics\\":{\\"average\\":500.0,\\"ips\\":null,\\"maximum\\":900,\\"median\\":450.0,\\"minimum\\":200,\\"mode\\":null,\\"percentiles\\":{\\"99\\":900},\\"sample_size\\":8,\\"std_dev\\":200.0,\\"std_dev_ips\\":null,\\"std_dev_ratio\\":0.4},\\"memory_usages\\":[200,400,400,400,500,500,700,900],\\"name\\":\\"My Job\\",\\"run_time_statistics\\":{\\"average\\":500.0,\\"ips\\":2.0e3,\\"maximum\\":900,\\"median\\":450.0,\\"minimum\\":200,\\"mode\\":400,\\"percentiles\\":{\\"99\\":900},\\"sample_size\\":8,\\"std_dev\\":200.0,\\"std_dev_ips\\":800.0,\\"std_dev_ratio\\":0.4},\\"run_times\\":[200,400,400,400,500,500,700,900]}]"
 
   """
-  @spec format(Suite.t(), map) :: %{Suite.key() => String.t()}
+  @spec format(Suite.t(), %{file: String.t()}) :: String.t()
   def format(%Suite{scenarios: scenarios}, %{file: _}) do
     scenarios
-    |> Enum.group_by(fn scenario -> scenario.input_name end)
-    |> Enum.map(fn {input, scenarios} ->
-      {input, format_scenarios_for_input(scenarios)}
+    |> Enum.map(fn scenario ->
+      Map.take(scenario, [
+        :name,
+        :input_name,
+        :run_time_statistics,
+        :memory_usage_statistics,
+        :run_times,
+        :memory_usages
+      ])
     end)
-    |> Map.new()
+    |> encode!()
   end
 
   def format(_, _) do
     raise """
     You need to specify a file to write the JSON to in the configuration a
     formatter option:
 
-      formatters: [{Benchee.Formatters.JSON, file: \"my.json\"]]"}]
+      formatters: [{Benchee.Formatters.JSON, file: \"my.json\"}]
     """
   end
 
   @doc """
   Uses the return value of `Benchee.Formatters.JSON.format/1` to write it to the
   JSON file defined in the initial configuration.
   """
-  @spec write(%{Suite.key() => String.t()}, map) :: :ok
+  @spec write(String.t(), map) :: :ok
   def write(data, %{file: file}) do
-    FileCreation.each(data, file)
-    :ok
-  end
-
-  # Do not make me private, poor HTML formatter relies on me
-  @doc false
-  @spec format_scenarios_for_input([Scenario.t()]) :: String.t()
-  def format_scenarios_for_input(scenarios) do
```

Comment:
this will wreak the HTML formatter but I guess we'll see how we need to deal with that, update it once we get that working again